en/services/component_information.md

```diff
@@ -88,42 +88,43 @@ Component information files may be **.xz** compressed (this is recommended for f
 
-> **Note** A component that supports this service must support the `COMP_METADATA_TYPE_VERSION` type and return a component file URI *that is hosted on the vehicle* (accessed using MAVLink FTP).
 
 In summary:
-1. Send `MAV_CMD_REQUEST_MESSAGE` specifying `param1=395` and `param2=COMP_METADATA_TYPE_VERSION`.
+1. GCS sends `MAV_CMD_REQUEST_MESSAGE` specifying `param1=395`.
    - This is a normal [command protocol](../services/command.md) request with timeouts and resends based on the ACK.
-1. The component will ACK the command and immediately send the requested `COMPONENT_INFORMATION` message (populated with URI and UIDs for the version information).
+1. The component will ACK the command and immediately send the requested `COMPONENT_INFORMATION` message (populated with URI and CRC for the general metadata file, and optionally with the URI and CRC for the non-MAVLink peripherals metadata file).
    - A `CMD_ACK` of anything other than `MAV_RESULT_ACCEPTED` indicates the protocol is not supported (sequence completes).
 1. GCS waits for the `COMPONENT_INFORMATION` message
    - If not recieved the GCS should resend the request (typically in the application level).
-   - Once information is received the GCS checks if `COMPONENT_INFORMATION.metadata_uid` matches cached component information.
-     If so, the sequence is **complete**.
-     Otherwise ...
-1. GCS downloads the file specified in the `metadata_uri` using MAVLink FTP (also the translation files, if any).
-1. GCS parses version metadata for `supportedCompMetadataTypes` and caches the results.
-1. The sequence above can be repeated to get the other supported data types (either immediately, or as needed).
+   - Once information is received:
+     - the GCS checks if `COMPONENT_INFORMATION.general_metadata_file_crc` matches its cached CRC value. If so, there is no need to download the general metadata file (or other files it references) as it has not changed since the last download.
+	 - the GCS checks if `COMPONENT_INFORMATION.peripherals_metadata_file_crc` matches its cached CRC value. If so, there is no need to download the peripherals metadata file (or other files it references).
+     If present it will also check the `COMPONENT_INFORMATION.peripherals_metadata_file_crc` matches the cached value.
+	 
+	 If the cached values do not match the associated files should be downloaded and parsed ....
+1. GCS downloads the file specified in the `general_metadata_uri` using MAVLink FTP.
+1. GCS parses the general metadata for other supported metadata locations, and then downloads the files via MAVFTP or HTTP (either immediately, or as needed).
 
 A GCS can discover all components in the system by monitoring the channel for `HEARTBEAT` ids, and then send the above request to each of them to verify whether the protocol is supported.
 
 Alternatively, a GCS can also broadcast the request; all components that support the protocol should respond with `COMPONENT_INFORMATION`.
```

Comment:
Might be clear, but also in this case the GCS needs to keep track about existing components on the link, because otherwise if a component does not receive the broadcast, it's seen as non-existent, whereas continous heartbeat check is more reliable (also for components that come online later).

Reply:
This is kind of a tricky one because in theory all components should emit a `HEARTBEAT` (with the exception of high latency links) and all nodes that can in theory route messages should maintain routing tables/info for all components. Of course QGC doesn't do this, and many components don't emit a heartbeat :-(, 

I have fixed this by adding a section on component discovery and putting the broadcast option first, and recommended.

Reply:
> many components don't emit a heartbeat

Really? That's quite broken then.